include/dynd/callable.hpp

```diff
@@ -155,9 +155,9 @@ namespace nd {
 
     std::intptr_t get_narg() const { return get_type()->get_npos(); }
 
-    const ndt::type &get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
+    const ndt::type get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
 
-    const array &get_arg_types() const { return get_type()->get_pos_types(); }
+    nd::array get_arg_types() const { return get_type()->get_pos_types(); }
```

Comment:
It would return a reference to a local object otherwise:

/home/stefan/libdynd/include/dynd/callable.hpp:160:53: error: returning reference to local temporary object [-Werror,-Wreturn-stack-address]
    const nd::array &get_arg_types() const { return get_type()->get_pos_types(); }